packages/vite/src/node/optimizer/index.ts

```diff
@@ -536,9 +538,20 @@ export function runOptimizeDeps(
     depsInfo,
     ssr,
     processingCacheDir,
+    optimizerContext,
   )
 
   const result = preparedRun.then(({ context, idToExports }) => {
+    function disposeContext() {
+      return context?.dispose().catch((e) => {
+        config.logger.error('error happed during context.dispose', e)
```

Comment:
```suggestion
        config.logger.error('Failed to dispose esbuild context', { error: e })
```

Looks like there's a typo, and the error isn't passed correctly.